pmd-java/src/main/resources/category/java/design.xml

```diff
@@ -927,7 +927,7 @@ Use opposite operator instead of negating the whole expression with a logic comp
             <property name="xpath">
                 <value>
 <![CDATA[
-//UnaryExpressionNotPlusMinus[@Image='!']/PrimaryExpression/PrimaryPrefix/Expression[EqualityExpression or RelationalExpression]
+//UnaryExpression[@Operator='!']/InfixExpression[@Operator = ('==', '<', '>', '<=', '>=')]
```

Comment:
shouldn't this also include `!=` testing for the double negative?
```suggestion
//UnaryExpression[@Operator='!']/InfixExpression[@Operator = ('==', '!=', '<', '>', '<=', '>=')]
```

Reply:
Probably yes, then we would be able to detect code like this:

```java
if (!(a != b)) { ...}  // instead of "a == b"
```

I'll try to add a test case for this.